openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/storages/builders/WaySurfaceTypeGraphStorageBuilder.java

```diff
@@ -48,39 +51,26 @@ public GraphExtension init(GraphHopper graphhopper) throws Exception {
 	public void processWay(ReaderWay way) {
 		waySurfaceDesc.reset();
 
-		boolean hasHighway = way.hasTag("highway");
-		boolean isFerryRoute = way.hasTag("route", ferries);
-
-		Iterator<Entry<String, Object>> it = way.getProperties();
-
-		if (isFerryRoute) {
-            waySurfaceDesc.setWayType((byte) WayType.FERRY);
-        }
-
-		while (it.hasNext()) {
-			Entry<String, Object> pairs = it.next();
-			String key = pairs.getKey();
-			String value = pairs.getValue().toString();
-
-			if (hasHighway || isFerryRoute) {
-				if (key.equals("highway")) {
-					byte wayType = (isFerryRoute) ? WayType.FERRY : (byte)WayType.getFromString(value);
+		int wayType;
+		if (way.hasTag(TAG_ROUTE, ferries)) {
+			wayType = WayType.FERRY;
+		} else if (way.hasTag(TAG_HIGHWAY)) {
+			wayType = WayType.getFromString(way.getTag(TAG_HIGHWAY));
+		} else {
+			return;
+		}
+		waySurfaceDesc.setWayType(wayType);
 
-					if (waySurfaceDesc.getSurfaceType() == 0) {
-						if (wayType == WayType.ROAD ||  wayType == WayType.STATE_ROAD || wayType == WayType.STREET) {
-                            waySurfaceDesc.setSurfaceType((byte) SurfaceType.ASPHALT);
```

Comment:
The assumed surface type `ASPHALT` is a more specific value than `PAVED` considered as the opposite of `UNPAVED`. I believe that if no `surface` tag is present  we should only stratify between the most general  complementing values `PAVED` and `UNPAVED`. This is backed up by to [OSM wiki](https://wiki.openstreetmap.org/wiki/Key:surface) stating: 
> For roads for motor vehicles there there is normally an assumption that the surface is `surface=paved` unless otherwise stated. Paved in OpenStreetMap is non-specific and may cover sealed, tarmac, asphalt, bitumen. `surface=unpaved` is treated as the opposite of paved.

Reply:
We will make the switch in a separate PR.